marathon/models/events.py

```diff
@@ -28,11 +26,7 @@ def __init__(self, event_type, timestamp, **kwargs):
         self.event_type = event_type  # All events have these two attributes
         self.timestamp = timestamp
         for attribute in self.KNOWN_ATTRIBUTES:
-            try:
-                self._set(attribute, kwargs[attribute])
-            except KeyError:
-                marathon.log.warn(
-                    'Unknown event attribute processing event {}: {}'.format(event_type, attribute))
+            self._set(attribute, kwargs.get(attribute))
```

Comment:
How about change the _set() method, if the attribute value is None then don't set the attribute, this will behavior as before only without warn log.